source/components/GlobalNavigation/components/User/MobileLoggedIn.spec.js

```diff
@@ -0,0 +1,117 @@
+import React, { useContext } from 'react';
+import { shallow } from 'enzyme';
+import ShallowRenderer from 'react-test-renderer/shallow';
+import merge from 'lodash/merge';
+
+import MobileLoggedIn from './MobileLoggedIn';
+
+const defaultProps = {
+    user: {
+        avatar_url: 'https://fandomdesignsystem.com/images/FANDOM-Avatar.jpg',
+        username: 'Ryba777',
+        'tracking-label': 'account',
+        items: [
+            {
+                type: 'link-text',
+                title: {
+                    type: 'translatable-text',
+                    key: 'global-navigation-user-view-profile',
+                },
+                href: '//starwars.wikia.com/wiki/User:Ryba777',
+                'tracking-label': 'account.profile',
+            },
+            {
+                type: 'link-text',
+                title: {
+                    type: 'translatable-text',
+                    key: 'global-navigation-user-my-talk',
+                },
+                href: '//starwars.wikia.com/wiki/User_talk:Ryba777',
+                'tracking-label': 'account.talk',
+            },
+            {
+                type: 'link-text',
+                title: {
+                    type: 'translatable-text',
+                    key: 'global-navigation-user-my-preferences',
+                },
+                href: '//starwars.wikia.com/wiki/Special:Preferences',
+                'tracking-label': 'account.preferences',
+            },
+            {
+                type: 'link-text',
+                title: {
+                    type: 'translatable-text',
+                    key: 'global-navigation-user-help',
+                },
+                href: '//community.wikia.com/wiki/Help:Contents',
+                'tracking-label': 'account.help',
+            },
+            {
+                type: 'link-logout',
+                href: 'https://www.wikia.com/logout',
+                title: {
+                    type: 'translatable-text',
+                    key: 'global-navigation-user-sign-out',
+                },
+                'param-name': 'returnto',
+                'tracking-label': 'account.sign-out',
+            },
+        ],
+    },
+    openModal: () => null,
+    track: () => null,
+    isOpen: false,
+};
+
+function renderComponent(props) {
+    const renderer = new ShallowRenderer();
+    const computedProps = merge({}, defaultProps, props);
+
+    renderer.render(<MobileLoggedIn {...computedProps} />);
+
+    return renderer.getRenderOutput();
+}
+
+// TODO: remove mocks when enzyme/react-test-renderer will fully support hooks
+jest.mock('react', () => ({
```

Comment:
I've tried multiple times to get this done, including jest's `doMock/mock` variations, re-exporting hooks from different module (and mocking it), mocking `react` globally or using other testing libs like [react-testing-library](https://github.com/kentcdodds/react-testing-library). and I still find this version the cleanest of them all.

I've tried to extract it to a separate method, but unfortunatelly `jest` sets it's context per test file, so it simply didn't work the way I expected as context was (I assume) lost at some point.